### PR TITLE
Test 'Bucket.list_blobs' w/ 'user_project' set.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: storage/tests/system.py [hunk 1]
```diff
@@ -360,6 +360,15 @@ def test_list_files(self):
         self.assertEqual(sorted(blob.name for blob in all_blobs),
                          sorted(self.FILENAMES))
 
+    @unittest.skipUnless(USER_PROJECT, 'USER_PROJECT not set in environment.')
+    @RetryErrors(unittest.TestCase.failureException)
+    def test_list_files_with_user_project(self):
+        with_user_project = Config.CLIENT.bucket(
+            self.bucket.name, user_project=USER_PROJECT)
+        all_blobs = list(with_user_project.list_blobs())
```
Comment (dhermes) [inline storage/tests/system.py]: This isn't very useful unless we add blobs first?

Comment (tseaver) [reply storage/tests/system.py]: Done in `setupClass`.

Comment (dhermes) [reply storage/tests/system.py]: Ah yes, I recall that now.

Comment (tseaver) [reply storage/tests/system.py]: Any remaining issues?

Comment (dhermes) [reply storage/tests/system.py]: Nope LGTM

**Pull request merged**